fdbserver/ClusterController.actor.cpp

```diff
@@ -742,130 +816,113 @@ class ClusterControllerData {
 		return results;
 	}
 
+	// Allows the comparison of two different recruitments to determine which one is better
+	// Tlog recruitment is different from all the other roles, in that it avoids degraded processes
+	// And tried to avoid recruitment in the same DC as the cluster controller
 	struct RoleFitness {
 		ProcessClass::Fitness bestFit;
 		ProcessClass::Fitness worstFit;
 		ProcessClass::ClusterRole role;
 		int count;
-		bool worstIsDegraded;
+		int worstUsed = 1;
+		bool degraded = false;
+		bool inClusterControllerDC = false;
 
 		RoleFitness(int bestFit, int worstFit, int count, ProcessClass::ClusterRole role)
 		  : bestFit((ProcessClass::Fitness)bestFit), worstFit((ProcessClass::Fitness)worstFit), count(count),
-		    role(role), worstIsDegraded(false) {}
+		    role(role) {}
 
 		RoleFitness(int fitness, int count, ProcessClass::ClusterRole role)
-		  : bestFit((ProcessClass::Fitness)fitness), worstFit((ProcessClass::Fitness)fitness), count(count), role(role),
-		    worstIsDegraded(false) {}
+		  : bestFit((ProcessClass::Fitness)fitness), worstFit((ProcessClass::Fitness)fitness), count(count),
+		    role(role) {}
 
 		RoleFitness()
 		  : bestFit(ProcessClass::NeverAssign), worstFit(ProcessClass::NeverAssign), role(ProcessClass::NoRole),
-		    count(0), worstIsDegraded(false) {}
-
-		RoleFitness(vector<WorkerDetails> workers, ProcessClass::ClusterRole role) : role(role) {
+		    count(0) {}
+
+		RoleFitness(const vector<WorkerDetails>& workers,
+		            ProcessClass::ClusterRole role,
+		            const std::map<Optional<Standalone<StringRef>>, int>& id_used,
+		            Optional<Standalone<StringRef>> ccDcId)
+		  : role(role) {
+			// Every recruitment will attempt to recruit the preferred amount through GoodFit,
+			// So a recruitment which only has BestFit is not better than one that has a GoodFit process
 			worstFit = ProcessClass::GoodFit;
-			worstIsDegraded = false;
+
+			degraded = false;
+			inClusterControllerDC = false;
 			bestFit = ProcessClass::NeverAssign;
+			worstUsed = 1;
```

Comment:
Consider removing the assignments that are already being assigned the same default value.